### ISSUE TITLE
Improve installer GUI

### ISSUE THREAD

**Issue by peters** [fintermobilityas/snapx#91]
**Is your feature request related to a problem? Please describe.**
Because snapx has only been used internally at our company the gif animation was just a funny thing introduced when we started development. In order to get other companies to adopt snapx in their commercial products this is probably a requirement. 

**Describe the solution you'd like**
![snapxinstaller](https://user-images.githubusercontent.com/869/86280198-a3e2c080-bbdb-11ea-99bb-c857dc948530.gif)

**Describe alternatives you've considered**
None.

**Additional context**
https://github.com/fintermobilityas/snapx/tree/issue-91

**Comment by peters** [comment]
Branch: https://github.com/fintermobilityas/snapx/tree/issue-91
![image](https://user-images.githubusercontent.com/869/86251178-d11a7900-bbb1-11ea-807e-73fd3095f136.png)

**Comment by hutterm** [comment]
so this is closed, but is it possible yet to change the gui?

**Comment by peters** [comment]
@hutterm Unfortunately, no. It's pretty hard-coded at the moment: https://github.com/fintermobilityas/snapx/blob/1bddfdfcd6b942c1e6be50552fd958ed3e5acf8f/src/Snap.Installer/ViewModels/AvaloniaMainWindowViewModel.cs#L91 👍

**Comment by hutterm** [comment]
hmm, this is preventing me pretty much to adopt this project.

I tried to get the build running, but I've already spent way too much time on figuring this out and getting everything to compile. So I'd just like to leave this here as a request:

The way I'd imagine this working would look about like this:
- Pull out the IMainWindowViewModel either into a separate project or into the snapx 
  - name this interface something IUpdateProgress
  - add a void `Progress(int p)` for progress reporting
- for a different UI I can then implement this interface, put it in a separate dll and tell snapx when packing to use that via a configuration setting in the yml
- so for the installer to do this he'd instantiate that class and run the progress and status texts through that interface

that'd be the easiest and probably fastest way I'd imagine. 

would that be an option?
